flycheck.el

```diff
@@ -2878,13 +2878,39 @@ variables of Flycheck."
 
 
 ;;; Errors from syntax checks
+(cl-defstruct (flycheck-coords
+               (:constructor flycheck-coords-new (start-line end-line start-col end-col)))
+  "Structure representing the location of an error as lines and columns."
+  start-line end-line start-col end-col)
+
+(cl-defstruct (flycheck-region
+               (:constructor flycheck-region-new (start-pos end-pos)))
+  "Structure representing the location of an error as byte-offsets from the start of a file."
+  start-pos end-pos)
+
+(cl-defun flycheck-region-from-cons ((start-pos . end-pos))
+  "Create a region from a cons cell of `(START-POS . END-POS)`."
+  (flycheck-region-new start-pos end-pos))
```

Comment:
I think the usual representation of ranges in Emacs is a simple cons.  Unless we need something fancier, it might be better to stick with that?